### ISSUE TITLE
Let's get this on CPAN!

### ISSUE THREAD

**Issue by zmughal** [jmcveigh/p5-graphics-vtk#1]
I like the work you've done to move this forward. I hadn't pushed this to CPAN yet because I was waiting to write XS wrappers. But that will take a while and the Inline::Python code is surprisingly robust.

I would like to merge your changes in and add neat things like cross platform testing. It needs a little clean up though. What do you think?

**Comment by jmcveigh** [comment]
Well, I'm unsure.  I would like to complete the exercises first.  More development to the framework will happen naturally as the exercises are completed,

**Comment by jmcveigh** [comment]
Also, Matt S. Trout of #perl on MAGNet says the AUTOLOAD is the wrong way to go about it.

**Comment by jmcveigh** [comment]
I'm happy to have you merge my changes.  Would you like to see more on this?  Would this project have possible remuneration?  

I have no customers requiring 3D graphics at this time for my small-town Perl / Fedora job shop so this is a volunteer effort right now.

**Comment by zmughal** [comment]
Yeah, I understand that adding more VTK code to test it would be helpful to make sure we cover all the cases. I was thinking about porting the examples from the [VTK wiki](http://www.vtk.org/Wiki/VTK/Examples) (also found in the [VTK source](https://github.com/Kitware/VTK/tree/master/Examples)).

Right, AUTOLOAD was a hack I had put together to make things work at first and I like what you are doing so far. I am planning on looking at it more closely some time this week once I free up some time.

Same here regarding the volunteer time. I have to use this as part of my research and I am not getting paid for it either. ;-) I usually write this kind of code in C/C++, Tcl, or Python, but I want to see Perl support for it as well.

**Comment by jmcveigh** [comment]
I write Perl well.  Perl 5 is my language of choice.  So you are aware, I have useful and pointed experience with C/C++ and SWIG.

**Comment by jmcveigh** [comment]
Well, I suppose this is a worthwhile use of my time.  I am able to do some more exercises.  We'll see where it goes from there.  The exercises are from the VTK Wiki.  They are in the folder titled demos.

**Comment by jmcveigh** [comment]
In the case you require Perl 5 or Fedora work to be done on your behalf I am really very good and seeking to perform work in these technologies for pay.

**Comment by jmcveigh** [comment]
Ok.  I completed 3 examples from the Wiki and it prompted a major change to the AUTOLOAD subroutine that added proper arguments to the constructors as well as supporting classes (such as mutable and immutable).  So I believe the Graphics::VTK namespace will be nurtured as the Wiki examples are completed with success.

**Comment by jmcveigh** [comment]
Could you STAR this project on GitHub, please?

**Comment by jmcveigh** [comment]
Thanks.  I'm glad you put your STAR on the project.

**Comment by jmcveigh** [comment]
I'm working with Class::MOP and Class::MOP::Class.  The Inline::Python import statement does a lot of the work on behalf of the Graphics::VTK module.  It is possible to avoid both the Proxy written in Python and the AUTOLOAD subroutine.  I'm looking into this at this time.  I've also completed 2 more examples from the VTK Wiki.

**Comment by jmcveigh** [comment]
There is a man on the Inline::Python mailing list that recommended using the up-to-date TCL bindings for VTK and Perl as they go together like peas and carrots.  I'm looking into meta programming with Moose using the TCL bindings.
